### DISCUSSION TITLE
What are the plans for --build-option?

### DISCUSSION THREAD

**Discussion by lrq3000** [pypa/setuptools#3882] [Q&A]
`--build-option` and `--global-option` are going to be [removed](https://github.com/pypa/pip/issues/11859) in pip 23.3 and already deprecated as of pip 23.1. This is not surprising given pip is a frontend, and these are specific flags for setuptools.

But what are setuptools plans with these for the long term? Are they going to be kept or are they going to get deprecated at some point?

**Comment by abravalheri** [comment]
Hi @lrq3000, you should be able to have somewhat backwards compatible approach with [pip's `--config-settings`](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-config-settings) or [build's `--config-setting` (or simply `-C`)](https://pypa-build.readthedocs.io/en/latest/?badge=latest#python--m-build---config-setting).

Please note however that these "config settings" will be passed to all steps in the build process according to PEP 517 (which includes the `sdist`). If you want to pass `wheel` or `sdist` specific parameters, maybe you could consider using [`build`](https://pypa-build.readthedocs.io/en/latest/?badge=latest). Please see an example below:


```bash
cd /tmp
mkdir -p myproj/src/mypkg
cd myproj
touch src/mypkg/__init__.py
touch src/mypkg/api.py
touch src/mypkg/cli.py
cat <<EOF >pyproject.toml
[build-system]
requires = ["setuptools"]
build-backend = "setuptools.build_meta"
EOF
python3 -m venv .venv
.venv/bin/python3 -m pip install -U pip build
.venv/bin/python -m build --wheel "-C--global-option=--verbose --no-user-cfg" "-C--build-option=--build-number 1b --compression deflated"
# ...
# Successfully built mypkg-0.0.0-1b-py3-none-any.whl
.venv/bin/python -m pip install dist/mypkg-0.0.0-1b-py3-none-any.whl
# ...
# Successfully installed mypkg-0.0.0
```
